src/topologies/halfedge.jl

```diff
@@ -144,126 +144,116 @@ function HalfEdgeTopology(elems::AbstractVector{<:Connectivity}; sort=true)
 
   # sort elements to make sure that they
   # are traversed in adjacent-first order
-  adjelems = sort ? adjsort(elems) : elems
-  eleminds = sort ? indexin(adjelems, elems) : 1:length(elems)
+  elemsort = sort ? adjsort(elems) : elems
+  eleminds = sort ? indexin(elemsort, elems) : 1:length(elems)
+  adjelems::Vector{Vector{Int}} = map(collect ∘ indices, elemsort)
 
   # start assuming that all elements are
-  # oriented consistently as CCW
-  CCW = trues(length(adjelems))
+  # oriented consistently (e.g. CCW)
+  isreversed = falses(length(adjelems))
 
-  # initialize with first element
+  # initialize half-edges using first element
   half4pair = Dict{Tuple{Int,Int},HalfEdge}()
-  elem = first(adjelems)
-  inds = collect(indices(elem))
-  v = CircularVector(inds)
-  n = length(v)
-  for i in 1:n
-    half4pair[(v[i], v[i + 1])] = HalfEdge(v[i], eleminds[1])
+  elem = first(eleminds)
+  inds = first(adjelems)
+  n = length(inds)
+  for i in eachindex(inds)
+    u = inds[i]
+    v = inds[mod1(i + 1, n)]
+    # insert halfedges u -> v and v -> u
+    he = get!(() -> HalfEdge(u, elem), half4pair, (u, v))
+    half = get!(() -> HalfEdge(v, nothing), half4pair, (v, u))
```

Comment:
Is there a reason why you prefer `get!` with this anonymous function instead of simply using `half4pair[(u,v)] = HalfEdge(u, elem)` for example?

Reply:
Directly `setindex`-ing will silently overwrite previously defined keys; I believe some of the issues I was encountering were due to that. The alternative to avoid overwriting is an additional `haskey` check, which does incur a computational cost (albeit, somewhat minor).

An different use of `get!` is like so `get!(half4pair, (u, v), HalfEdge(u, elem))`, which looks slightly cleaner, but also has a performance penalty, because the default element is created even if not used, unlike with the anonymous function, which is only called if needed.